qtcodes/fitters/repetition.py

```diff
@@ -20,88 +20,48 @@ class RepetitionDecoder(LatticeDecoder):
-            self.node_map["Z"][vnode] = self.S["Z"].add_node(vnode)
+    def _params_validation(self):
+        if "phase-flip-protected" not in self.params:
+            self.params["phase-flip-protected"] = False
 
-        edge_weight = 1
-        for t in range(0, self.params["T"]):
-            # real nodes
-            for j in range(0, self.params["d"] - 1):
-                node = (t, j + 0.5, 0)
-                self.node_map["Z"][node] = self.S["Z"].add_node(node)
+        if self.params["phase-flip-protected"]:
+            self.syndrome_graph_keys = ["X"]
 
-            # edges (real-real)
-            for j in range(1, self.params["d"] - 1):
-                left = (t, j - 0.5, 0)
-                right = (t, j + 0.5, 0)
-                self.S["Z"].add_edge(
-                    self.node_map["Z"][left], self.node_map["Z"][right], edge_weight
+        # The repetition code is a 1D lattice
+        if "d" not in self.params:
+            if self.params["phase-flip-protected"]:
+                self.params["d"] = (1, 3)
+            else:
+                self.params["d"] = (3, 1)
+        elif isinstance(self.params["d"], Number):
+            d = int(self.params["d"])
+            if self.params["phase-flip-protected"]:
+                self.params["d"] = (1, d)
+            else:
+                self.params["d"] = (d, 1)
+        elif isinstance(self.params["d"], Tuple):
+            if (
+                not self.params["phase-flip-protected"]
+                and self.params["d"][constants.DW] != 1
+            ):
+                raise LatticeError(
+                    "Repetition qubits can only have width 1 in parameter d: e.g. (3,1)."
                 )
-
-            # edges (real-virtual)
-            self.S["Z"].add_edge(
-                self.node_map["Z"][self.virtual["Z"][0]],
-                self.node_map["Z"][(t, 0.5, 0)],
-                edge_weight,
-            )  # left
-            self.S["Z"].add_edge(
-                self.node_map["Z"][self.virtual["Z"][1]],
-                self.node_map["Z"][(t, self.params["d"] - 1.5, 0)],
-                edge_weight,
-            )  # right
-
-        # connect physical qubits in same location across subgraphs of adjacent times
-        syndrome_nodes_t0 = [(t, x, y) for t, x, y in self.S["Z"].nodes() if t == 0]
-        for node in syndrome_nodes_t0:
-            space_label = (node[1], node[2])
-            for t in range(0, self.params["T"] - 1):
-                self.S["Z"].add_edge(
-                    self.node_map["Z"][(t,) + space_label],
-                    self.node_map["Z"][(t + 1,) + space_label],
-                    edge_weight,
+            if (
+                self.params["phase-flip-protected"]
+                and self.params["d"][constants.DH] != 1
+            ):
+                raise LatticeError(
+                    "Phase-flip protected repetition qubits can only have height 1 in parameter d: e.g. (1,3)."
```

Comment:
I may have to go the common route because of how decoders are initialized. The encoder is only instantiated when the `.encoder` property is accessed, so this would cause param validation to fail later down the line. We could instantiate the encoder in `__init__` if you want to share the validation. Let me know what you think @Phionx.

Reply:
Can we make this validation a static method in the encoder class?

Reply:
Sure, I can try that.